### PR TITLE
[CS2] Nodes: Fix immediately invoked named classes

### LINKED ISSUE #4436: [CS2] Nodes: Compilation of named classes in expressions is broken
```coffee
new class
# new class {}

new class A
# new (A = )class A {};
```

### PR DESCRIPTION
**Pull request by connec**
Fixes #4436.

### CODE REVIEW

File: src/nodes.coffee [hunk 1]
```diff
@@ -1301,20 +1301,19 @@ exports.Class = class Class extends Base
...
     if @variable
-      assign = new Assign @variable, new Literal(''), null, { @moduleDeclaration }
-      [ assign.compileToFragments(o)..., result... ]
-    else
-      result
+      node = new Assign @variable, node, null, { @moduleDeclaration }
+
+    @compileNode = @compileClassDeclaration
+    try
+      return node.compileToFragments o
```
Comment (GeoffreyBooth) [inline src/nodes.coffee]: Why is this in a `try` block? When would it ever throw an exception, and in what cases would we want to catch such an exception?

Comment (connec) [inline src/nodes.coffee]: This is just ensure the override to `compileNode` is undone. We don't want to catch it all (hence no `catch`).

**Comment by connec** [comment]
Sorry for the force pushes @GeoffreyBooth, had to update authorship (reinstalled my OS :cry:).

File: src/nodes.coffee (hunk 1 above)
Comment (connec) [inline src/nodes.coffee]: It's possible that we should have a separate `ClassDeclaration` node, which would avoid these shenanigans. That feels like a bit too much of a refactor for this issue though.

Comment (GeoffreyBooth) [inline src/nodes.coffee]: Okay, I trust you.

**Comment by GeoffreyBooth** [comment]
@lydell, any notes on this? Looks good to me.

**Comment by lydell** [comment]
I guess it’s fine.

**Pull request merged**